File: CS/DXSpellCheckerBindingDictionaries/ViewModel/MainViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Xpf.SpellChecker;
using DevExpress.XtraSpellChecker;
using System;
using System.Globalization;

namespace DXSpellCheckerBindingDictionaries.ViewModel
{
    public class MainViewModel
    {
        public virtual DictionarySourceCollection Dictionaries { get; set; }
        public virtual SpellChecker SpellChecker { get; set; }
        protected virtual IMessageBoxService MessageBoxService { get { return null; } }

        public MainViewModel()
        {
            // Create a SpellChecker class instance.
            SpellChecker = new SpellChecker();
            // Specify the SpellChecker's properties.
            SpellChecker.SpellingFormType = SpellingFormType.Word;
            SpellChecker.SpellCheckMode = SpellCheckMode.AsYouType;
            SpellChecker.Culture = new CultureInfo("de-DE");
            // Obtain a dictionary collection.
            Dictionaries = GetDictionaries();

            // Subscribe to events
            SpellChecker.RepeatedWordFound += SpellChecker_RepeatedWordFound;
            SpellChecker.CheckCompleteFormShowing += SpellChecker_CheckCompleteFormShowing;

        }

        private void SpellChecker_CheckCompleteFormShowing(object sender, FormShowingEventArgs e)
        {
            MessageBoxService.Show("That's It!");
            e.Handled = true;
        }

        private void SpellChecker_RepeatedWordFound(object sender, RepeatedWordFoundEventArgs e)
        {
            e.Result = SpellCheckOperation.Cancel;
        }

        public DictionarySourceCollection GetDictionaries()
        {
            // Create a dictionary collection.
            var collection = new DictionarySourceCollection();
            
            // Create a Hunspell spell-checking dictionary.
            var dictionary = new HunspellDictionarySource() { Culture = new CultureInfo("de-DE") };

            // Load the dictionary file.
            dictionary.DictionaryUri = new Uri(@"pack://application:,,,/DXSpellCheckerBindingDictionaries;component/Dictionaries/de_DE.dic");
            // Load the affix file.
            dictionary.GrammarUri = new Uri(@"pack://application:,,,/DXSpellCheckerBindingDictionaries;component/Dictionaries/de_DE.aff");
            // Add the dictionary to the collection.
            collection.Add(dictionary);

            var customDictionary = new SpellCheckerCustomDictionarySource() { Culture = new CultureInfo("EN-us") };
            customDictionary.DictionaryUri = new Uri(@"pack://application:,,,/DXSpellCheckerBindingDictionaries;component/Dictionaries/CustomEnglish.dic");
            customDictionary.AlphabetUri = new Uri(@"pack://application:,,,/DXSpellCheckerBindingDictionaries;component/Dictionaries/Alphabet.txt");
            collection.Add(customDictionary);
            return collection;
        }
    }
}

File: VB/DXSpellCheckerBindingDictionaries/ViewModel/MainViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Xpf.SpellChecker
Imports DevExpress.XtraSpellChecker
Imports System.Globalization

Namespace DXSpellCheckerBindingDictionaries.ViewModel

    Public Class MainViewModel

        Public Overridable Property Dictionaries As DictionarySourceCollection

        Public Overridable Property SpellChecker As SpellChecker

        Protected Overridable ReadOnly Property MessageBoxService As IMessageBoxService
            Get
                Return Nothing
            End Get
        End Property

        Public Sub New()
            ' Create a SpellChecker class instance.
            SpellChecker = New SpellChecker()
            ' Specify the SpellChecker's properties.
            SpellChecker.SpellingFormType = SpellingFormType.Word
            SpellChecker.SpellCheckMode = SpellCheckMode.AsYouType
            SpellChecker.Culture = New CultureInfo("de-DE")
            ' Obtain a dictionary collection.
            Dictionaries = GetDictionaries()
            ' Subscribe to events
            Me.SpellChecker.RepeatedWordFound += AddressOf SpellChecker_RepeatedWordFound
            Me.SpellChecker.CheckCompleteFormShowing += AddressOf SpellChecker_CheckCompleteFormShowing
        End Sub

        Private Sub SpellChecker_CheckCompleteFormShowing(ByVal sender As Object, ByVal e As FormShowingEventArgs)
            MessageBoxService.Show("That's It!")
            e.Handled = True
        End Sub

        Private Sub SpellChecker_RepeatedWordFound(ByVal sender As Object, ByVal e As RepeatedWordFoundEventArgs)
            e.Result = SpellCheckOperation.Cancel
        End Sub

        Public Function GetDictionaries() As DictionarySourceCollection
            ' Create a dictionary collection.
            Dim collection = New DictionarySourceCollection()
            ' Create a Hunspell spell-checking dictionary.
            Dim dictionary = New HunspellDictionarySource() With {.Culture = New CultureInfo("de-DE")}
            ' Load the dictionary file.
            dictionary.DictionaryUri = New Uri("pack://application:,,,/DXSpellCheckerBindingDictionaries;component/Dictionaries/de_DE.dic")
            ' Load the affix file.
            dictionary.GrammarUri = New Uri("pack://application:,,,/DXSpellCheckerBindingDictionaries;component/Dictionaries/de_DE.aff")
            ' Add the dictionary to the collection.
            collection.Add(dictionary)
            Dim customDictionary = New SpellCheckerCustomDictionarySource() With {.Culture = New CultureInfo("EN-us")}
            customDictionary.DictionaryUri = New Uri("pack://application:,,,/DXSpellCheckerBindingDictionaries;component/Dictionaries/CustomEnglish.dic")
            customDictionary.AlphabetUri = New Uri("pack://application:,,,/DXSpellCheckerBindingDictionaries;component/Dictionaries/Alphabet.txt")
            collection.Add(customDictionary)
            Return collection
        End Function
    End Class
End Namespace
